### PR TITLE
Initial compatibility info for WeakRef and FinalizationQueue

### PR DESCRIPTION
**Pull request by tjcrowder**
My first try adding compat info from scratch, apologies for any rough edges.

Chrome version is based on https://www.chromestatus.com/features/5892186633666560
Firefox behind-a-flag version is based on the milestone here: https://bugzilla.mozilla.org/show_bug.cgi?id=1593698
Firefox by default version based on https://hg.mozilla.org/mozilla-central/rev/8cf72e804176

I've verified the Firefox preference in v77.

### CODE REVIEW

**Comment by tjcrowder** [comment]
@Elchi3 - Thanks! Done.

File: javascript/builtins/FinalizerRegistry.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+{
+  "javascript": {
+    "builtins": {
+      "FinalizerRegistry": {
```
Comment (Elchi3) [inline javascript/builtins/FinalizerRegistry.json:4]:
```suggestion
      "FinalizationRegistry": {
```

File: javascript/builtins/FinalizerRegistry.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/FinalizerRegistry",
```
Comment (Elchi3) [inline javascript/builtins/FinalizerRegistry.json:6]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry",
          "spec_url": "https://tc39.es/proposal-weakrefs/#sec-finalization-registry-objects",
```

File: javascript/builtins/WeakRef.json [hunk 3]
```diff
@@ -0,0 +1,74 @@
+{
+  "javascript": {
+    "builtins": {
+      "WeakRef": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WeakRef",
```
Comment (Elchi3) [inline javascript/builtins/WeakRef.json:6]:
```suggestion
          "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WeakRef",
          "spec_url": "https://tc39.es/proposal-weakrefs/#sec-weak-ref-objects",
```

**Comment by tjcrowder** [comment]
@Elchi3 - Man, I really was having an off day, wasn't I? Thank you for all of that, and for being kind about it.

I'll add the subfeatures. I thought the breakdown was only done when there was variability. But A) Am I wrong about that? And B) There *is* variability, because the support data is by environment, not engine; `cleanupSome` is likely not to be supported on web browsers, at least not initially, even though it's supported by the engines in them. (I have a note on the [method itself](https://wiki.developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry/cleanupSome).)

So, off to see what the reality on the ground is...

**Comment by Elchi3** [comment]
No worries, happy to help. :slightly_smiling_face: 

> I thought the breakdown was only done when there was variability

We always break things down to make compatibility explicit. There are way too often noteworthy things for sub features of an API or just varying support as you say. So, this granularity is always required in mdn-browser-compat-data.

Thanks for going the extra miles!

**Comment by tjcrowder** [comment]
Some notes on the latest:

* Firefox for Android is v68 and I'm told by Jon Coppeard at Mozilla it always will be since they're doing Firefox Preview now instead. So "firefox_android" is false, can't be mirrored from "firefox" anymore.
* Opera is currently shipping on a Chromium v81 base. I couldn't find any public signals about when they'll update to v84 and thus get weakrefs.
* The proposal doesn't link to anything about JavaScriptCore having support yet. Couldn't find any public signals about Safari, so for now, went with `false`.

Thank you again @Elchi3 for your help and patience thus far!

File: javascript/builtins/WeakRef.json [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+          },
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        },
+        "WeakRef": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WeakRef/WeakRef",
```
Comment (Elchi3) [inline javascript/builtins/WeakRef.json:63]:
```suggestion
            "description": "<code>WeakRef()</code> constructor",
            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/WeakRef/WeakRef",
```

File: javascript/builtins/FinalizerRegistry.json [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+          },
+          "status": {
+            "experimental": false,
+            "standard_track": true,
+            "deprecated": false
+          }
+        },
+        "FinalizationRegistry": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry/FinalizationRegistry",
```
Comment (Elchi3) [inline javascript/builtins/FinalizerRegistry.json:63]:
```suggestion
            "description": "<code>FinalizationRegistry()</code> constructor",
            "mdn_url": "https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry/FinalizationRegistry",
```

**Comment by tjcrowder** [comment]
@Elchi3 - Thanks again for all your work walking me through this. :-) Next time it should be a lot smoother!

**Comment by tjcrowder** [comment]
@Elchi3 - It looks like the compat data still isn't on the [page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakRef). Did I do something wrong, or is there a pipeline issue...?

**Comment by Wshao62** [comment]
Hi


Do you have any sales data?  Mailbox

T.J. Crowder <notifications@github.com> 於 2020年6月13日週六 17:23 寫道：

**Comment by Elchi3** [comment]
@tjcrowder pages had to be re-rendered. I just did that, using force-refresh while logged in on the wiki page. All good now.

**Pull request merged**